datalad/distribution/metadata.py

```diff
@@ -0,0 +1,116 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Interface for managing metadata
+"""
+
+__docformat__ = 'restructuredtext'
+
+
+from datalad.interface.base import Interface
+from datalad.interface.common_opts import recursion_limit
+from datalad.distribution.dataset import datasetmethod, EnsureDataset
+from ..support.param import Parameter
+from ..support.constraints import EnsureStr, EnsureBool, EnsureNone
+from ..log import lgr
+
+
+class MetaData(Interface):
+    """Manage dataset metadata.
+
+    Datalad's native metadata for a dataset is a flat list of term/value pairs
+    (similar to DataDryad). There is no limit on the number and nature of
+    metadata terms, although datalad itself only understands a few terms.
+    Moreover, datalad supports multiple categories or sets of metadata. This is
+    useful when managing hand-curated information in addition to automatically
+    extracted metadata.
+
+    The order in which operations are executed is:
+
+    1. Removal
+    2. Replacement
+    3. Addition
+    4. Aggregation
+    5. Query
+
+    Metadata items and terms can be addressed in various ways:
+
+    - by index (run `get *` to see items indexed): -> "#16"
```

Comment:
it refers to the 16th item in the list of meta data items, which you'd see from `get *`